src/Type/Php/RegexArrayShapeMatcher.php

```diff
@@ -45,7 +48,20 @@ public function __construct(
 	{
 	}
 
+	public function matchExpr(Expr $patternExpr, ?Type $flagsType, TrinaryLogic $wasMatched, Scope $scope): ?Type
```

Comment:
maybe it would even make sense to use `Expr` also for `$flags`, just in case for more flexibility for future enhancements?

Reply:
hmm.. having `$flags` as `Expr` would make it harder for composer/pcre to [pass the internal flags](https://github.com/composer/pcre/pull/29/files#diff-f885e19d2e2b46c78de6962b8e1c48ad6c556f056da18e02ecf4cdb9713ff428R33). 

It seems we should leave it as is in this PR